core/src/lib.rs

```diff
@@ -93,6 +94,28 @@ pub struct CoreInitOptions {
     pub telemetry_opts: TelemetryOptions,
 }
 
+/// TODO:  This holds an actual worker now. Lang keeps track of taskq->worker mapping
+pub struct WorkerImpl {}
+
+/// Initialize a worker bound to a task queue
+pub fn init_worker<SG: WorkflowServiceClientTrait + Send + Sync + 'static>(
+    _worker_config: WorkerConfig,
+    _client: SG,
+) -> WorkerImpl {
+    todo!()
+}
+
+/// Create a worker for replaying a specific history. It will auto-shutdown as soon as the history
+/// has finished being replayed. The provided gateway should be a mock, and this should only be used
+/// for workflow testing purposes.
+pub fn init_replay_worker(
+    config: WorkerConfig,
+    gateway: Arc<dyn ServerGatewayApis + Send + Sync>,
+    history: &History,
```

Comment:
Would we need the gateway here?
I think we can just create the gateway based on history.

Reply:
I can certainly add a convenience function for it. This level of control is useful for my integration tests (hence why it needs to be public, but can be doc-hidden), because they might want to ensure certain rpc calls are or are not made.